tfhe/src/shortint/parameters/mod.rs

```diff
@@ -131,11 +118,195 @@ impl Parameters {
     }
 }
 
+#[derive(Serialize, Copy, Clone, Deserialize, Debug, PartialEq)]
+pub enum ShortintParameterSetInner {
+    PBSOnly(PBSParameters),
+    WopbsOnly(WopbsParameters),
+    PBSAndWopbs(PBSParameters, WopbsParameters),
+}
+
+#[derive(Serialize, Copy, Clone, Deserialize, Debug, PartialEq)]
+pub struct ShortintParameterSet {
+    inner: ShortintParameterSetInner,
+}
+
+impl ShortintParameterSet {
+    pub const fn new_pbs_param_set(params: &PBSParameters) -> Self {
```

Comment:
I would take input params by ownership and not ref here

Reply:
any reason for that ? I think "large" structs are better passed by ref ?

Reply:
Yes but I would say its only if you don't copy them, here we take a ref, to then copy what it points to, why not take a copy directly ?

Reply:
fair enough

Reply:
gen_keys from integer takes params by ref while gen_keys from shortint takes params by value, could be the occasion to chose one over the other

Reply:
normally done and uniformized integers

Reply:
I'd vote to take by value :sweat_smile: but its not the end of the world if the other way is chosen

Reply:
by value is what I ended up using 🙂